include/cell.tcc

```diff
@@ -450,9 +469,17 @@ inline bool mpm::Cell<Tdim>::is_point_in_cell(
   else
     (*xi) = this->transform_real_to_unit_cell(point);
 
-  // Check if the transformed coordinate is within the unit cell (-1, 1)
-  for (unsigned i = 0; i < (*xi).size(); ++i)
-    if ((*xi)(i) < -1. || (*xi)(i) > 1. || std::isnan((*xi)(i))) status = false;
+  // Check if the transformed coordinate is within the unit cell:
+  // between 0 and 1-xi(1-i) if the element is a triangle, and between
+  // -1 and 1 if otherwise
+  switch (this->element_->corner_indices().size()) {
```

Comment:
Why can't you have an `if else`?

Reply:
I thought about it. But if in the future someone wants to implement another type of element than this structure would be more helpful. Should I still change it to if-else?